packages/core/src/external/carequality/command/documents/send-ihe-gateway-results.ts

```diff
@@ -6,84 +6,94 @@ import duration from "dayjs/plugin/duration";
-const TABLE_NAME = "document_query_result";
 
 const api = axios.create();
 
 dayjs.extend(duration);
 
-export const DOC_QUERY_TIMEOUT = dayjs.duration({ minutes: 3 });
+export const CONTROL_TIMEOUT = dayjs.duration({ minutes: 3 });
 const CHECK_DB_INTERVAL = dayjs.duration({ seconds: 5 });
 
 type RaceControl = { isRaceInProgress: boolean };
 
-export const sendDocumentQueryResults = async ({
+export const sendIHEGatewayResults = async <TableResult>({
   requestId,
   patientId,
   cxId,
-  numOfLinks,
+  numOfGateways,
   dbCreds,
   endpointUrl,
+  resultsTable,
 }: {
   requestId: string;
   patientId: string;
   cxId: string;
-  numOfLinks: number;
+  numOfGateways: number;
   dbCreds: string;
   endpointUrl: string;
+  resultsTable: string;
 }): Promise<void> => {
   const sequelize = initSequelizeForLambda(dbCreds);
 
   const raceControl: RaceControl = { isRaceInProgress: true };
 
   try {
-    // Run the document query until it either times out, or all the results are in the database
+    // Run the table count until it either times out, or all the results are in the database
     const raceResult = await Promise.race([
       controlDuration(),
-      checkNumberOfResults(sequelize, raceControl, requestId, numOfLinks),
+      checkNumberOfResults(sequelize, raceControl, resultsTable, requestId, numOfGateways),
     ]);
 
-    const documentQueryResults = await queryDocumentQueryResults(sequelize, requestId);
+    const iheGatewayResults = await queryIHEGatewayResults<TableResult>(
+      sequelize,
+      resultsTable,
+      requestId
+    );
 
     if (raceResult) {
       console.log(
-        `${raceResult}. Got ${documentQueryResults.length} successes out of ${numOfLinks} gateways for PD. RequestID: ${requestId}`
+        `${raceResult}. Got ${iheGatewayResults.length} successes out of ${numOfGateways} gateways for ${resultsTable}. RequestID: ${requestId}`
       );
       raceControl.isRaceInProgress = false;
     }
 
-    await api.post(`${endpointUrl}/internal/carequality/document-query/results`, {
+    await api.post(endpointUrl, {
       requestId,
       patientId,
       cxId,
-      documentQueryResults,
+      iheGatewayResults,
     });
   } catch (error) {
-    const msg = `Failed to post document query results`;
+    const msg = `Failed to post ihe gateway results - table: ${resultsTable}`;
     console.log(`${msg} - endpoint ${endpointUrl}. Error: ${errorToString(error)}`);
-    throw new MetriportError(msg, error, { requestId, patientId, cxId, numOfLinks });
+    throw new MetriportError(msg, error, { requestId, patientId, cxId, numOfGateways });
   }
 };
 
 async function controlDuration(): Promise<string> {
-  const timeout = DOC_QUERY_TIMEOUT.asMilliseconds();
+  const timeout = CONTROL_TIMEOUT.asMilliseconds();
   await sleep(timeout);
-  return `Document Query reached timeout after ${timeout} ms`;
+  return `IHE gateway reached timeout after ${timeout} ms`;
 }
 
 async function checkNumberOfResults(
```

Comment:
nit: also could be reused from a shared location with a callback to check for completion